nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -0,0 +1,71 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.keycloak;
+in {
+  options = {
+    services.keycloak = {
+      enable = mkEnableOption "Keycloak identity and access server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keycloak;
+        defaultText = "pkgs.keycloak";
+        example = literalExample "pkgs.keycloak";
+        description = "Package that should be used for keycloak";
+      };
+
+      jbossBaseDir = mkOption {
+        type = types.str;
+        default = "/var/lib/keycloak";
+        description = ''
+          The location keycloak (and underlying JBoss/Wildfly) information
+          (configuration, logs, data) is stored.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        description = "The user to run keycloak as.";
+        default = "keycloak";
+      };
+
+      group = mkOption {
+        type = types.str;
+        description = "The group to run keycloak as.";
+        default = "keycloak";
+      };
+
+      extraFlags = mkOption {
```

Comment:
Every option requires a `type`. The type here should probably be a list. You should change the `default = "";` to `default = [];` to reflect this.

Reply:
fixed